kong/plugins/opentelemetry/utils.lua

```diff
@@ -48,8 +58,117 @@ local function get_headers(conf_headers)
 end
 
 
+local compile_opts = {
+  escape = "\xff", -- disable '#' as a valid template escape
+}
+
+local template_cache = setmetatable( {}, { __mode = "k" })
+
+
+local function param_value(source_template, resource_attributes, template_env)
+  if not source_template or source_template == "" then
+    return nil
+  end
+
+  if not lua_enabled then
+    -- Detect expressions in the source template
+    local expr = str_find(source_template, "%$%(.*%)")
+    if expr then
+      return nil, "loading of untrusted Lua code disabled because " ..
+                  "'untrusted_lua' config option is set to 'off'"
+    end
+    -- Lua is disabled, no need to render the template
+    return source_template
+  end
+
+  -- find compiled templates for this plugin-configuration array
+  local compiled_templates = template_cache[resource_attributes]
+  if not compiled_templates then
+    compiled_templates = {}
+    -- store it by `config_array` which is part of the plugin `conf` table
```

Comment:
caching looks good here. Please update the comment which still refers to the original code (mentioning config_array).